src/pcnt.rs

```diff
@@ -0,0 +1,543 @@
+    #[doc = "     - EspError"]
+    pub fn filter_disable(&self) -> Result<(), EspError> {
+        unsafe { esp!(esp_idf_sys::pcnt_filter_disable(self.unit)) }
+    }
+
+    #[doc = " @brief Set PCNT filter value"]
+    #[doc = ""]
+    #[doc = " @param filter_val PCNT signal filter value, counter in APB_CLK cycles."]
+    #[doc = "        Any pulses lasting shorter than this will be ignored when the filter is enabled."]
+    #[doc = "        @note"]
+    #[doc = "        filter_val is a 10-bit value, so the maximum filter_val should be limited to 1023."]
+    #[doc = ""]
+    #[doc = " @return"]
+    #[doc = "     - ()"]
+    #[doc = "     - EspError"]
+    pub fn set_filter_value(&self, value: u16) -> Result<(), EspError> {
+        unsafe { esp!(esp_idf_sys::pcnt_set_filter_value(self.unit, value)) }
+    }
+
+    #[doc = " @brief Get PCNT filter value"]
+    #[doc = ""]
+    #[doc = " @return"]
+    #[doc = "     - i16"]
+    #[doc = "     - EspError"]
+    pub fn get_filter_value(&self) -> Result<u16, EspError> {
+        let mut value = 0u16;
+        unsafe {
+            esp!(esp_idf_sys::pcnt_get_filter_value(
+                self.unit,
+                &mut value as *mut u16
+            ))?;
+        }
+        Ok(value)
+    }
+
+    #[doc = " @brief Set PCNT counter mode"]
+    #[doc = ""]
+    #[doc = " @param channel PCNT channel number"]
+    #[doc = " @param pos_mode Counter mode when detecting positive edge"]
+    #[doc = " @param neg_mode Counter mode when detecting negative edge"]
+    #[doc = " @param hctrl_mode Counter mode when control signal is high level"]
+    #[doc = " @param lctrl_mode Counter mode when control signal is low level"]
+    #[doc = ""]
+    #[doc = " @return"]
+    #[doc = "     - ()"]
+    #[doc = "     - EspError"]
+    pub fn set_mode(&self,
+        channel: PcntChannel,
+        pos_mode: PcntCountMode,
+        neg_mode: PcntCountMode,
+        hctrl_mode: PcntControlMode,
+        lctrl_mode: PcntControlMode) -> Result<(), EspError> {
+        unsafe {
+            esp!(pcnt_set_mode(self.unit, channel.into(), pos_mode.into(), neg_mode.into(), hctrl_mode.into(), lctrl_mode.into()))
+        }
+    }
+
+    #[doc = " @brief Add ISR handler for specified unit."]
+    #[doc = ""]
+    #[doc = " This ISR handler will be called from an ISR. So there is a stack"]
+    #[doc = " size limit (configurable as \"ISR stack size\" in menuconfig). This"]
+    #[doc = " limit is smaller compared to a global PCNT interrupt handler due"]
+    #[doc = " to the additional level of indirection."]
+    #[doc = ""]
+    #[doc = " @param callback Interrupt handler function."]
+    #[doc = ""]
+    #[doc = " @return"]
+    #[doc = "     - ()"]
+    #[doc = "     - EspError"]
+    pub unsafe fn subscribe(&self, callback: impl FnMut(u32) + Send + 'static) -> Result<(), EspError> {
+        enable_isr_service()?;
+
+        //self.unsubscribe();
+        let callback: Box<dyn FnMut(u32) + 'static> = Box::new(callback);
```

Comment:
In no_std, `Box` is not part of prelude. You need to import it from the alloc crate, or use `alloc::boxed::Box`.